File: iterator/java/IteratorTest.java
public class IteratorTest {
    public static void main(String[] args) {
        MyList<String> list = new MyArrayList<>();
        list.add("San Zhang");
        list.add("Si Li");
        list.add("Wu Wang");

        MyIterator<String> itr = list.iterator();
        while (itr.hasNext()) {
            String element = itr.next();
            System.out.println(">>> 当前元素：" + element);
            itr.remove();
            // list.remove(1);
        }
        System.out.println("[Array List] --> " + list);

        System.out.println("==========================================");

        list = new MyLinkedList<>();
        list.add("San Zhang");
        list.add("Si Li");
        list.add("Wu Wang");

        itr = list.iterator();
        while (itr.hasNext()) {
            String element = itr.next();
            System.out.println(">>> 当前元素：" + element);
            itr.remove();
            // list.remove(2);
        }
        System.out.println("[Linked List] --> " + list);
    }
}

File: iterator/go/iterator_test.go
package iterator

import (
	"testing"
)

func TestIterator(t *testing.T) {
	var list MyList[string] = NewMyArrayList[string]()
	list.Add("San Zhang")
	list.Add("Si Li")
	list.Add("Wu Wang")

	itr := list.Iterator()
	for itr.HasNext() {
		element, err := itr.Next()
		if err != nil {
			t.Fatal(err)
		}
		t.Logf(">>> 当前元素：%s", element)
		err = itr.Remove()
		// err = list.Remove(1)
		if err != nil {
			t.Fatal(err)
		}
	}
	t.Logf("[Array List] --> %s", list)

	t.Log("==========================================")

	list = NewMyLinkedList[string]()
	list.Add("San Zhang")
	list.Add("Si Li")
	list.Add("Wu Wang")

	itr = list.Iterator()
	for itr.HasNext() {
		element, err := itr.Next()
		if err != nil {
			t.Fatal(err)
		}
		t.Logf(">>> 当前元素：%s", element)
		err = itr.Remove()
		// err = list.Remove(2)
		if err != nil {
			t.Fatal(err)
		}
	}
	t.Logf("[Linked List] --> %s", list)
}
